Improve wafer map die highlighting API (#1701)
ni/nimble: packages/nimble-components/src/wafer-map/specs/README.md

    -   `dieLabelsHidden` - a boolean value that determines if the die labels in the wafer map view are shown or not. Default value is false.
    -   `dieLabelsSuffix` - represent a string that can be added as a label in the end of the each data information in the wafer map dies value.
    -   `colorScaleMode` - represent an Enum value that determent if the colorScale is represent a continues gradient values (linear), or is set categorically (ordinal).

>     -   `highlightedTags` - represent a list of strings based on which the highlight will be done

Reviewer: ```suggestion
    -   `highlightedTags` - a list of strings that will denote which dies to highlight
```

    -   `disabled` - it's represented by a boolean value and refers to the state of the `nimble-wafer-map` component. If true, the component should be rendered dimmed out and no user interaction should be allowed.
    -   `validity` - readonly object of boolean values that represents the validity states that the wafer map's configuration can be in. The object's type is `WaferMapValidity`, and it contains the following boolean properties:
        -   `invalidGridDimensions` : true when any of the `gridMinX`, `gridMinY`, `gridMaxX` or `gridMaxY` is `undefined`, but false when all of them are `undefined`.

The `originLocation`, `orientation`, `gridMinX`, `gridMinY`, `gridMaxX`, `gridMaxY`, `dieCharacterCount`, `disabled`, `waferDataType` and `colorBy` properties will be configurable via properties and attributes.

> The `dies`, `colorScale` and `highlightedTags` properties will be configurable only via properties and will not have attributes.

Author: You are right, I added "tags?: string[]" at line 92 (where the WaferMapDie is displayed), and I expended the documentation to be more clear about how tags should be used, at 110
